scratchapi2/front.py

```diff
@@ -16,93 +16,115 @@ class FrontPage
 from .user import Project, User
 from .gclass import GenericData
 
+GETINFO = False
+
 class FrontPage(object):
-    """ A class to handle front page """
+    """The Front Page of the Scratch website."""
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """Enforce this class as a singleton."""
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance

```

Comment:
This specially-crafted `__new__` classmethod is used to make sure that the `FrontPage` class has exactly one instance ever. Same for other singleton classes.